src/relay/pass/dead_code.cc

```diff
@@ -110,7 +110,11 @@ class CalcDep : private ExprVisitor {
   VarMap<size_t> use_map_;
 
   void VisitExpr(const Expr& e) final {
-    return ExprFunctor<void(const Expr& e)>::VisitExpr(e);
+    visit_counter_[e.get()]++;
+    if (visit_counter_[e.get()] <= 2) {
```

Comment:
why `<=2`?

Reply:
the dce code seprate variable into three parts:
used 0 times (remove)
used 1 times (inline)
used 2 times (dont do anything).

Reply:
okay, let's add a comment

Reply:
Done.